### ISSUE TITLE
[Bug]: Puppeteer hangs indefinitely in a browser that has previously encountered reCAPTCHA and CloudFlare turnstile captcha

### ISSUE THREAD

**Issue by Gabriel-Bigelow** [puppeteer/puppeteer#12715]
### Steps to reproduce

My github repo with minimum reproducible example: https://github.com/Gabriel-Bigelow/puppeteer_hanging_after_encountering_recaptcha_or_cloudflare_turnstile

YouTube video with example of the bug: https://www.youtube.com/watch?v=ZtuKmGituj8
```
const puppeteer = require("puppeteer");

const checkIfPuppeteerHangs = async () => {
	console.log("connecting to puppeteer");
	console.log("trying to connect to browser");
	const browser = await puppeteer.connect({
		browserURL: "http://127.0.0.1:9222/json/version",
	});
	console.log("trying to open a new page");
	const page = await browser.newPage();

	console.log("getting existing pages");
	console.log(await browser.pages());

	console.log("We didn't get stuck. Disconnecting from old browser");
	await browser.disconnect();
	console.log("trying to connect to newBrowser");
	const newBrowser = await puppeteer.connect({
		browserURL: "http://127.0.0.1:9222/json/version",
	});
	console.log("connected to newBrowser. Getting pages...");
	const pages = await newBrowser.pages();
	console.log(pages);
};
checkIfPuppeteerHangs();
```

1. Open a Chrome remote debugging window
2. Visit Indeed.com
3. Apply to any job using the internal Indeed application process. (Click apply now on a job that does not link you away from Indeed.com)
4. Fill out any information fields and click the continue button until you reach the "/intervention" or "/review" page.
5. Start a puppeteer script that uses await `puppeteer.connect({ browserURL: "http://127.0.0.1:9222/json/version" })` or `await puppeteer.connect({browserWSEndpoint: whateverYourWSEndpointIs })`.

It will hang indefinitely. This will happen regardless of closing the tabs that are on Indeed.com before starting the Puppeteer script. I believe that the captcha system on Indeed.com may be permanently changing the current browser instance that causes Puppeteer to not be able to connect to it. 

### Expected results

Manually applying to jobs on Indeed.com before running a Puppeteer script does not affect the ability of Puppeteer to connect to the remote debugging window, create new pages, or get an array of the current pages.

### Actual results

Puppeteer hangs indefinitely at `await puppeteer.connect({browserURL: "http://127.0.0.1:9222/json/version" })` after manually reaching the "/review" page during the application process on Indeed.com

**Comment by OrKoN** [comment]
Could you please include the CDP log for the connect call? https://pptr.dev/guides/debugging#log-devtools-protocol-traffic If there are any targets that CDP fails to automatically attach and Puppeteer waits for them, you can work around the issue using the target filter.

P.S. This is the wrong category for the bug report.

**Comment by OrKoN** [comment]
If the site has some sort of bot detection and it is blocking targets using infinite loops and that sort of thing, then it’s out of scope for Puppeteer to work around such protections.

**Comment by crispycarrot** [comment]
https://github.com/puppeteer/puppeteer/issues/11627

**Comment by FlueschPluesch** [comment]
I could kinda reproduce this problem with a simple example. Just change the url to see the difference.
```
//const puppeteer = require('puppeteer-extra');
const puppeteer = require('puppeteer');
//const StealthPlugin = require('puppeteer-extra-plugin-stealth');
//puppeteer.use(StealthPlugin());

const { spawn } = require('child_process');

const CHROME_EXECUTABLE_PATH = 'C:/Program Files/Google/Chrome/Application/chrome.exe';
const USER_DATA_DIR = 'C:/Users/JB/AppData/Local/Google/Chrome/User Data';

// Function to launch Chrome with remote debugging
async function launchChrome() {
  // Path to your Chrome executable
  // Spawn a new Chrome process with remote debugging enabled
  const chromeProcess = spawn(CHROME_EXECUTABLE_PATH, [
    `--remote-debugging-port=9222`,
    `--user-data-dir=${USER_DATA_DIR}`
]);

  // Wait for a few seconds to ensure Chrome has started
  await new Promise(resolve => setTimeout(resolve, 1000));

  // Connect Puppeteer to the remote debugging port
  const browser = await puppeteer.connect({
    browserURL: 'http://127.0.0.1:9222',
  });

  // Open a new page and navigate to a URL
  const page = await browser.newPage();

//let url = 'https://example.com'; //Not protected
let url = 'https://dexscreener.com/'; //Cloudflare protected

  await page.goto(url);

  console.log('Page loaded!');

  // Close the browser after some time
  setTimeout(async () => {

    console.log('DC!');

    await browser.disconnect(); // Disconnect Puppeteer

    console.log('Trying to connect again!'); 

    const browser2 = await puppeteer.connect({ // Will be stuck here forever on cloudflare pages
        browserURL: 'http://127.0.0.1:9222',
      });

    console.log('Connected!');

    const page2 = await browser2.newPage();
    await page2.goto(url);

    //await browser.close();
    //chromeProcess.kill();

  }, 5000);
}

launchChrome().catch(console.error);
```

**Comment by crispycarrot** [comment]
Seeing this with latest `puppeteer-core` and Chrome on Debian and MacOS. Issue occurs after a captcha is encountered. 

The call to `Puppeteer.connect` hangs. Current workaround is `Promise.race` with `Puppeteer.connect` and `setTimeout`. If `setTimeout` returns, this implies a hung call to `Puppeteer.connect`, and the browser can be killed and restarted + detached from the parent process.

Similar problem occurs with `Page.goto` in which the frame ID is not found and navigation fails. Workaround currently is to iterate the page's frames to find the correct one and utilize that for navigation and subsequent `waitFor` and selector calls.

If I had to guess, based on #11627 and the filed Chrome issue, this is not a Puppeteer problem and is something that has occurred upstream in Chrome itself.

**Comment by OrKoN** [comment]
I tried to browser indeed.com but I could not reproduce the issue but it might be that I didn't encounter the right captcha. I also tried https://www.cloudflare.com/products/turnstile/ and I am still not able to reproduce. I was able to reproduce with the `https://dexscreener.com/` example though. I am not sure yet what the root cause is but the browser reports available workers with no url that never get auto-attached. A work-around: 

```
  const browser = await puppeteer.connect({
    browserURL: 'http://127.0.0.1:9222',
    targetFilter: (target) => {
        if (target.type() === 'other' && target.url() === '') {
            return false;
        }
        return true;
    }
  });  
```

It might be related to the linked crbug.com but I am not sure.
